runners/flink/src/main/java/org/apache/beam/runners/flink/translation/utils/FlinkPipelineTranslatorUtils.java

```diff
@@ -77,4 +77,25 @@ public static WindowingStrategy getWindowingStrategy(
           e);
     }
   }
+
+  public static String genOperatorNameFromStagePayload(
```

Comment:
This isn't really Flink-specific. I wonder if it'd be better placed somewhere in core-construction.

Reply:
Perhaps `ExecutableStageTranslation` would be a good place to put this.